### DISCUSSION TITLE
Just chatting

### DISCUSSION THREAD

**Discussion by maekawatoshiki** [maekawatoshiki/vicis#8] [General]
Anything about vicis is welcome here

**Comment by hsk** [comment]
設計の方針を書いてほしいです。
パーサーとASTとプリンターが混ざってるのがよく分からない。
支配辺境がAST周りに絡んでてよく分からないのですが、その辺の設計がなぜ今の形になっていて素晴らしいのかを説明してあるドキュメントがあると嬉しいように思います。

**Reply by maekawatoshiki** [reply]
> パーサーとASTとプリンターが混ざってるのがよく分からない。

全てがcore crateに含まれているのが良くないという意味でしょうか。

**Comment by ramasai1** [comment]
cool project, i would like to contribute but i don't know Rust -- wish it was in C++ haha. still a good excuse to learn Rust this way i guess!

**Reply by maekawatoshiki** [reply]
Thank you for being interested in! Try reading some tests and examples and hopefully create a small piece of code using vicis :)
